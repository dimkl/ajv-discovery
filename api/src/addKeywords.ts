import type Ajv from "ajv";

export const addKeywords = (ajv: Ajv) => {
  ajv.addKeyword({
    keyword: "http_method",
    validate: () => true,
    errors: false,
  });
  ajv.addKeyword({
    keyword: "http_path",
    validate: () => true,
    errors: false,
  });
};
