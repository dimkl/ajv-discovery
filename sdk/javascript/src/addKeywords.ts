import type Ajv from "ajv";

// see: /api/src/addKeywords.ts
export const addKeywords = (ajv: Ajv) => {
  ajv.addKeyword({
    keyword: "http_method",
    validate: () => true,
    errors: false,
  });
  ajv.addKeyword({
    keyword: "http_path",
    validate: () => true,
    errors: false,
  });
};
