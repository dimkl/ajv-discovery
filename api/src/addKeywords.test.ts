import { addKeywords } from "./addKeywords";

describe("addKeywords(ajv)", () => {
  test("sets http_method and http_path keywords in ajv passed", () => {
    const ajv = { addKeyword: jest.fn() } as any;

    addKeywords(ajv);

    expect(ajv.addKeyword).toBeCalledTimes(2);
    expect(ajv.addKeyword.mock.calls[0]).toMatchSnapshot();
    expect(ajv.addKeyword.mock.calls[0][0].validate()).toEqual(true);
    expect(ajv.addKeyword.mock.calls[1]).toMatchSnapshot();
    expect(ajv.addKeyword.mock.calls[1][0].validate()).toEqual(true)
  });
});
