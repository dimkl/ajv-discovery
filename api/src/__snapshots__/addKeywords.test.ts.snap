// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addKeywords(ajv) sets http_method and http_path keywords in ajv passed 1`] = `
[
  {
    "errors": false,
    "keyword": "http_method",
    "validate": [Function],
  },
]
`;

exports[`addKeywords(ajv) sets http_method and http_path keywords in ajv passed 2`] = `
[
  {
    "errors": false,
    "keyword": "http_path",
    "validate": [Function],
  },
]
`;
